### PR TITLE
[Impeller] Fix DrawPicture.

### PR DESCRIPTION
**Pull request by bdero**
Prior to this patch, `DrawPicture` was untested and a no-op.

We needed a new routine to absorb Elements of a cloned pass into another pass, since appending a `Picture` as a subpass would be incorrect behavior for drawing a picture (since subpasses start with a blank image, which gets drawn to and then composited with the parent pass via a separate blending operation).

### CODE REVIEW

**Review by dnfield — APPROVED** [review]

File: impeller/aiks/canvas.cc [hunk 1]
```diff
@@ -413,7 +416,9 @@ void Canvas::DrawPicture(Picture picture) {
                              entity.GetTransformation());
     return true;
   });
-  return;
+  pass->AddSubpassInline(std::move(pass));
```
Comment (dnfield) [inline impeller/aiks/canvas.cc:419]:
```suggestion
  GetCurrentPass()->AddSubpassInline(std::move(pass));
```

File: impeller/aiks/canvas.cc [hunk 2]
```diff
@@ -401,7 +401,10 @@ void Canvas::DrawPoints(std::vector<Point> points,
   GetCurrentPass().AddEntity(entity);
 }
 
-void Canvas::DrawPicture(Picture picture) {
+void Canvas::DrawPicture(const Picture& picture) {
+  auto save_count = GetSaveCount();
+  Save();
```
Comment (dnfield) [inline impeller/aiks/canvas.cc:406]: Move this below the early return below.

**Review by dnfield — CHANGES_REQUESTED** [review]
Uncaught yb linter issue

File: impeller/entity/entity_pass.cc [hunk 3]
```diff
@@ -840,6 +856,19 @@ bool EntityPass::OnRender(
   return true;
 }
 
+void EntityPass::IterateAllElements(
```
Comment (dnfield) [inline impeller/entity/entity_pass.cc:859]: Consider moving to separate patch, unused

File: impeller/aiks/canvas.cc (hunk 1 above)
Comment (bdero) [reply impeller/aiks/canvas.cc:419]: Haha, done.

File: impeller/aiks/canvas.cc (hunk 2 above)
Comment (bdero) [reply impeller/aiks/canvas.cc:406]: Done.

File: impeller/entity/entity_pass.cc (hunk 3 above)
Comment (bdero) [reply impeller/entity/entity_pass.cc:859]: Removed.

**Comment by chinmaygarde** [comment]
Is this part of the fixes for https://github.com/flutter/flutter/issues/130078?

**Comment by dnfield** [comment]
No this is me nerd-sniping @bdero into fixing this the right way haha

**Comment by dnfield** [comment]
I filed https://github.com/flutter/flutter/issues/130142 to track the work I'm referring to.
